pallets/liquidity-pools-gateway/queue/src/lib.rs

```diff
@@ -141,9 +135,9 @@ pub mod pallet {
 			let message = MessageQueue::<T>::take(nonce).ok_or(Error::<T>::MessageNotFound)?;
 
 			match Self::process_message_and_deposit_event(nonce, message.clone()) {
```

Comment:
Curios if there's a way to add the weight that we now return from this call to the total call weight.

Reply:
Maybe in the expansion of that we can check if it's correctly reverted

Reply:
``` impl<T: Config> frame_support::dispatch::GetDispatchInfo for Call<T> {
        fn get_dispatch_info(&self) -> frame_support::dispatch::DispatchInfo {
            match *self {
                Self::process_message { ref nonce } => {
                    let __pallet_base_weight = T::WeightInfo::process_message()
                        .saturating_add(T::process_message_and_deposit_event().1);
                    let __pallet_weight = <dyn frame_support::dispatch::WeighData<
                        (&T::MessageNonce,),
                    >>::weigh_data(&__pallet_base_weight, (nonce,));
                    let __pallet_class = <dyn frame_support::dispatch::ClassifyDispatch<
                        (&T::MessageNonce,),
                    >>::classify_dispatch(&__pallet_base_weight, (nonce,));
                    let __pallet_pays_fee = <dyn frame_support::dispatch::PaysFee<
                        (&T::MessageNonce,),
                    >>::pays_fee(&__pallet_base_weight, (nonce,));
                    frame_support::dispatch::DispatchInfo {
                        weight: __pallet_weight,
                        class: __pallet_class,
                        pays_fee: __pallet_pays_fee,
                    }
                }
```

Reply:
But it doesn't compile, gotta play with this for a bit.

Reply:
Using the standard weight calculation that we have in there, I ran:

```
target/release/centrifuge-chain benchmark pallet \
  --chain=development \
  --steps=50 \
  --repeat=20 \
  --pallet=pallet-liquidity-pools-gateway-queue \
  --extrinsic="*" \
  --wasm-execution=compiled \
  --heap-pages=4096 \
  --output=test.rs
```

and got [these weights](https://gist.github.com/cdamian/e1aa677e6e0046662a3caa28ef3be913)

Reply:
I think the PoV is enough in those weights 👍🏻